services/collaboration/pkg/connector/fileconnector.go

```diff
@@ -562,6 +549,34 @@ func (f *FileConnector) CheckFileInfo(ctx context.Context) (FileInfo, error) {
 		fileInfo.IsAnonymousUser = true
 	}
 
+	switch wopiContext.ViewMode {
+	case appproviderv1beta1.ViewMode_VIEW_MODE_READ_WRITE:
+		fileInfo.SupportsUpdate = true
+		fileInfo.UserCanWrite = true
+
+	case appproviderv1beta1.ViewMode_VIEW_MODE_READ_ONLY:
+		// nothing special to do here for now
+
+	case appproviderv1beta1.ViewMode_VIEW_MODE_VIEW_ONLY:
+		fileInfo.DisableExport = true
+		fileInfo.DisableCopy = true
+		fileInfo.DisablePrint = true
+		fileInfo.WatermarkText = f.ensureWatermarkText(fileInfo, wopiContext.User) //the feature supported by Collabora only
+	}
+
 	logger.Debug().Msg("CheckFileInfo: success")
 	return fileInfo, nil
 }
+
+func (f *FileConnector) ensureWatermarkText(fileInfo FileInfo, user *userv1beta1.User) string {
+	if fileInfo.IsAnonymousUser {
```

Comment:
Does it make sense to move this check out of this function? The fileinfo is "being built", so using something that hasn't fully built might not be a good idea. I mean, you had to move the code because the fileinfo didn't contain the final values for these properties.
There is also no comment that prevents me to overwrite the `UserFriendlyName` property (for example) just before returning the fileinfo, and that would be a bug because the watermark wouldn't use the right information.

My point is that I don't expect any particular order when setting attributes of an object. I should be able to do it in any random order and always get the same result.